test/performance/odp_ipsec.c

```diff
@@ -0,0 +1,1035 @@
+	memset(mem + sizeof(test_data), 1, payload_length - sizeof(test_data));
+
+	return pkt;
+}
+
+/**
+ * Run measurement iterations for given config and payload size.
+ * Result of run returned in 'result' out parameter.
+ */
+static int
+run_measure_one(ipsec_args_t *cargs,
+		ipsec_alg_config_t *config ODP_UNUSED,
+		odp_ipsec_sa_t sa,
+		unsigned int payload_length,
+		crypto_run_result_t *result)
+{
+	odp_ipsec_out_param_t param;
+	odp_pool_t pkt_pool;
+	odp_queue_t out_queue;
+	odp_packet_t pkt = ODP_PACKET_INVALID;
+	int rc = 0;
+
+	pkt_pool = odp_pool_lookup("packet_pool");
+	if (pkt_pool == ODP_POOL_INVALID) {
+		app_err("pkt_pool not found\n");
+		return -1;
+	}
+
+	out_queue = odp_queue_lookup("ipsec-out");
+	if (cargs->schedule || cargs->poll) {
+		if (out_queue == ODP_QUEUE_INVALID) {
+			app_err("ipsec-out queue not found\n");
+			return -1;
+		}
+	}
+
+	if (cargs->reuse_packet) {
+		pkt = make_packet(pkt_pool, payload_length);
+		if (ODP_PACKET_INVALID == pkt)
+			return -1;
+	}
+
+	time_record_t start, end;
+	int packets_sent = 0;
+	int packets_received = 0;
+
+	/* Initialize parameters block */
+	memset(&param, 0, sizeof(param));
+	param.num_sa = 1;
+	param.num_opt = 0;
+	param.sa = &sa;
+
+	fill_time_record(&start);
+
+	while ((packets_sent < cargs->iteration_count) ||
+	       (packets_received < cargs->iteration_count)) {
+		if ((packets_sent < cargs->iteration_count) &&
+		    (packets_sent - packets_received <
+		     cargs->in_flight)) {
+			odp_packet_t out_pkt;
+
+			if (!cargs->reuse_packet) {
+				pkt = make_packet(pkt_pool, payload_length);
+				if (ODP_PACKET_INVALID == pkt)
+					return -1;
+			}
+
+			out_pkt = cargs->in_place ? pkt : ODP_PACKET_INVALID;
+
+			if (cargs->debug_packets)
+				odp_packet_print_data(pkt, 0,
+						      odp_packet_len(pkt));
+
+			if (cargs->schedule || cargs->poll) {
+				rc = odp_ipsec_out_enq(&pkt, 1,
+						       &param);
+				if (rc <= 0) {
+					app_err("failed odp_crypto_packet_op_enq: rc = %d\n",
+						rc);
+					if (!cargs->reuse_packet)
+						odp_packet_free(pkt);
+					break;
+				}
+				packets_sent += rc;
+			} else {
+				int num_out = 1;
+
+				rc = odp_ipsec_out(&pkt, 1,
+						   &out_pkt, &num_out,
+						   &param);
+				if (rc <= 0) {
+					app_err("failed odp_ipsec_out: rc = %d\n",
+						rc);
+					if (!cargs->reuse_packet)
+						odp_packet_free(pkt);
+					break;
+				}
+				if (odp_packet_has_error(out_pkt)) {
+					odp_ipsec_packet_result_t result;
+
+					odp_ipsec_result(&result, out_pkt);
+					app_err("Received error packet: %d\n",
+						result.status.error.all);
+				}
+				packets_sent += rc;
+				packets_received += num_out;
+				if (cargs->debug_packets)
+					odp_packet_print_data(out_pkt, 0,
+							      odp_packet_len(out_pkt));
+				if (cargs->reuse_packet)
+					pkt = out_pkt;
+				else
+					odp_packet_free(out_pkt);
+			}
+		}
+
+		if (cargs->schedule || cargs->poll) {
+			odp_event_t ev;
+			odp_crypto_packet_result_t result;
```

Comment:
do not name with the same name as function arg parameter. Nothing bad but different names are easy to maintain.